### ISSUE TITLE
Adding tooltips to connections/arrows themselves?

### ISSUE THREAD

**Issue by FarhadGSRX** [terrastruct/d2#1260]
As the subject line says, is it possible for us to add tooltip/hover capability to arrows or connections themselves?

My use case is creating UML Diagrams in which the arrows themselves encode information in a tooltip, such as a description of the data object being passed from one class to another (which would otherwise be too unwieldy to just place as a label).

Thanks in advance!

**Comment by ferrieux** [comment]
I concur. This would be a huge improvement, as for long connections the position of the "central" label is often very far, and one needs to painstakingly "follow by eye" in order to find the relevant information. Arrowheads only partially solve the issue, but a tooltip is way superior, as it allows to pack more information than would fit in a permanent label.

Moreover, a quick experiment editing the SVG by hand shows it is pretty easy:
```xml
 <path> ... <title>Foo</title></path>
```

In short, that'd be great and looks like low-hanging fruit :)
